starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -1445,8 +1441,8 @@ sub check_star_catalog {
         my @dr;
         if ($type =~ /GUI|BOT/){
 	    foreach my $pixel (@bad_pixels) {
-		my $dy = abs($yag-$pixel->{yag});
-		my $dz = abs($zag-$pixel->{zag});
+		my $dy = abs($pixel_row-$pixel->{row}) * 5;
```

Comment:
For this comparison, was the previous logic incorrect?  We convert the bad pixels to yag, zag and confirm yag zag of those pixels not within dither + 25 arcsecs of star position.  Multiplying the pixel values by 5 is not that close to the previous values.  I'm wondering if it would make sense to leave this alone for this release and then either update to match the proseco logic (work in pixel space instead) or put that bad pixel check in sparkles (to be integrated in starcheck #327 ).

Reply:
The previous logic was correct and I believe the new logic is also correct. Note that the scale factor is being applied to a delta row/col, so the worst case change is likely < 0.05 arcsec on a check of 16 + 25 arcsec. From the perspective of actual ACA operational requirements that level of difference is inconsequential. Multiplying by the mean scale factor (around 4.996?) would reduce that even more.

My only concern is if this can cause a mismatch with sparkles. But at first glance it doesn't look like sparkles checks that, so that probably can't happen. And it sounds like you know that proseco works in pixel space, so then this change is already accomplishing a good thing, no?

Reply:
I missed that this was a delta pixel space check in this version and was just seeing pixels * 5.  I'm blaming the lack of whitespace around an arithmetic operator.

Reply:
Interesting, so I am changing my expectations of variations in the local plate scale with this:
```
In [9]: y1, z1 = pixels_to_yagzag(-300, -300)
In [10]: y2, z2 = pixels_to_yagzag(-300, -310)
In [11]: (z1 - z2) / 10
Out[11]: 4.9470113374697124
```
So what matters to me really is whether starcheck can fail a catalog that proseco selected as OK. I'm not quite sure off hand. @jeanconn maybe you have been looking at the code and the threshold that proseco applies? 

At the end of the day there is a fundamental point that bad pixels are defined in pixel space and stars are located in yag/zag space, so no matter how you define the distance metric there will be changes depending on the local plate scale.

Reply:
Right.  With regard to bad pixels defined in pixel space and stars in yag zag space, agreed.  I was just thinking that the converter mention that 'center' is often more what you have in mind and for this check that is based on "star distance to a pixel" the distance to pixel center is probably more right. 

But to your point about how consistent this is with proseco, the guide star exclusion for bad pixels is a little opaque (sorry) https://github.com/sot/proseco/blob/7a75ae329a3fde4070fbe808633db2abdcf19344/proseco/guide.py#L1330 .  The candidate star positions are converted to row/col with edge https://github.com/sot/proseco/blob/master/proseco/core.py#L1253 .  It looks to me like we're adding 4 to 5 pixels padding to define the exclusion range, so I would think this could be 1 pixel less conservative than the starcheck check for some cases.  

So worst case would be that the starcheck check would give us a red warning and we'd need to double-check this.  I was thinking this should probably go as-is and we can re-examine the tolerances in the future if they cause issues.

Reply:
>  I was thinking this should probably go as-is and we can re-examine the tolerances in the future if they cause issues.

YES!